interop-testing/src/main/java/io/grpc/testing/integration/CustomBackendMetricsLoadBalancerProvider.java

```diff
@@ -0,0 +1,182 @@
+    private final Helper helper;
+    private Subchannel subchannel;
+
+    public CustomBackendMetricsLoadBalancer(Helper helper) {
+      this.helper = OrcaOobUtil.newOrcaReportingHelper(checkNotNull(helper, "helper"));
+    }
+
+    @Override
+    public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
+      List<EquivalentAddressGroup> servers = resolvedAddresses.getAddresses();
+      if (subchannel == null) {
+        final Subchannel subchannel = helper.createSubchannel(
+            CreateSubchannelArgs.newBuilder()
+                .setAddresses(servers)
+                .build());
+        subchannel.start(new SubchannelStateListener() {
+          @Override
+          public void onSubchannelState(ConnectivityStateInfo stateInfo) {
+            processSubchannelState(subchannel, stateInfo);
+          }
+        });
+        this.subchannel = subchannel;
+        helper.updateBalancingState(CONNECTING,
+            new MayReportLoadPicker(PickResult.withSubchannel(subchannel)));
+        subchannel.requestConnection();
+      } else {
+        subchannel.updateAddresses(servers);
+      }
+    }
+
+    private void processSubchannelState(Subchannel subchannel, ConnectivityStateInfo stateInfo) {
+      ConnectivityState currentState = stateInfo.getState();
+      if (currentState == SHUTDOWN) {
+        return;
+      }
+      if (stateInfo.getState() == TRANSIENT_FAILURE || stateInfo.getState() == IDLE) {
+        helper.refreshNameResolution();
+      }
+
+      SubchannelPicker picker;
+      switch (currentState) {
+        case IDLE:
+          subchannel.requestConnection();
+          picker = new MayReportLoadPicker(PickResult.withNoResult());
+          break;
+        case CONNECTING:
+          picker = new MayReportLoadPicker(PickResult.withNoResult());
+          break;
+        case READY:
+          picker = new MayReportLoadPicker(PickResult.withSubchannel(subchannel));
+          break;
+        case TRANSIENT_FAILURE:
+          picker = new MayReportLoadPicker(PickResult.withError(stateInfo.getStatus()));
+          break;
+        default:
+          throw new IllegalArgumentException("Unsupported state:" + currentState);
+      }
+      helper.updateBalancingState(currentState, picker);
+    }
+
+    @Override
+    public void handleNameResolutionError(Status error) {
+    }
+
+    @Override
+    public void shutdown() {
+      if (subchannel != null) {
+        subchannel.shutdown();
+      }
+    }
+
+    private final class MayReportLoadPicker extends SubchannelPicker {
+      private PickResult result;
+
+      public MayReportLoadPicker(PickResult result) {
+        this.result = checkNotNull(result, "result");
+      }
+
+      @Override
+      public PickResult pickSubchannel(PickSubchannelArgs args) {
+        if (result.getSubchannel() == null) {
+          return result;
+        }
+        OrcaOobUtil.setListener(result.getSubchannel(), new OrcaOobUtil.OrcaOobReportListener() {
```

Comment:
No, I don't think we want to set it on a per-RPC basis. We should set the listener outside the picker and just store the most recent OrcaLoadReport into the RPC.